propertyestimator/protocols/simulation.py

```diff
@@ -353,34 +512,117 @@ def execute(self, directory, available_resources):
 
         # Create the yank yaml input file from a dictionary of options.
         with open(yaml_filename, 'w') as file:
-            yaml.dump(self._get_yank_options(), file)
+            yaml.dump(self._get_full_input_dictionary(directory), file)
 
-        exp_builder = ExperimentBuilder(yaml_filename)
-        exp_builder.setup_experiments()
+        with self._temporarily_change_directory(directory):
 
-        exp_builder.run_experiments()
+            exp_builder = ExperimentBuilder('yank.yaml')
+            exp_builder.run_experiments()
 
         return self._get_output_dictionary()
 
 
 @register_calculation_protocol()
-class RunHoseGuestYankProtocol(BaseProtocol):
-    """An abstract base class for protocols which will performs a set of alchemical
-    free energy simulations using the YANK framework.
+class LigandReceptorYankProtocol(BaseYankProtocol):
+    """An abstract base class for protocols which will performs a set of
+    alchemical free energy simulations using the YANK framework.
 
-    Protocols which inherit from this base must implement the abstract `_get_yank_options`
-    methods.
+    Protocols which inherit from this base must implement the abstract
+    `_get_*_dictionary` methods.
     """
 
-    def _get_yank_options(self):
-        """Returns a dictionary of options which will be serialized
-        to a yaml file and passed to YANK.
+    @protocol_input(str)
+    def ligand_residue_name(self):
+        """The residue name of the ligand."""
+        pass
+
+    @protocol_input(str)
+    def solvated_ligand_coordinates(self):
+        """The file path to the solvated ligand coordinates."""
+        pass
+
+    @protocol_input(str)
+    def solvated_ligand_system(self):
+        """The file path to the solvated ligand system object."""
+        pass
+
+    @protocol_input(str)
+    def solvated_complex_coordinates(self):
+        """The file path to the solvated complex coordinates."""
+        pass
+
+    @protocol_input(str)
+    def solvated_complex_system(self):
+        """The file path to the solvated complex system object."""
+        pass
+
+    def __init__(self, protocol_id):
+        """Constructs a new LigandReceptorYankProtocol object."""
+
+        super().__init__(protocol_id)
+
+        self._ligand_residue_name = None
+
+        self._solvated_ligand_coordinates = None
+        self._solvated_ligand_system = None
+
+        self._solvated_complex_coordinates = None
+        self._solvated_complex_system = None
+
+        self._local_ligand_coordinates = 'ligand.pdb'
```

Comment:
If you're using the Yank Python API, you shouldn't need to go through `.pdb` and `.xml` files---unless this is more convenient!
http://getyank.org/latest/api/yank_api.html

Reply:
For now this is more convenient for how things are set up as a whole - but I'll look into switching over to this in future!